src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsalertsapi/resource/CreateAlertCodeIntTest.kt

```diff
@@ -0,0 +1,374 @@
+package uk.gov.justice.digital.hmpps.hmppsalertsapi.resource
+
+import org.assertj.core.api.Assertions
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.test.web.reactive.server.WebTestClient
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.integration.IntegrationTestBase
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.integration.wiremock.TEST_USER
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.integration.wiremock.USER_NOT_FOUND
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.integration.wiremock.USER_THROW_EXCEPTION
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.AlertCode
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.model.request.CreateAlertCodeRequest
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.repository.AlertCodeRepository
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.repository.AlertTypeRepository
+import uk.gov.justice.digital.hmpps.hmppsalertsapi.utils.alertTypeVulnerability
+import uk.gov.justice.hmpps.kotlin.common.ErrorResponse
+import java.time.temporal.ChronoUnit
+
+class CreateAlertCodeIntTest : IntegrationTestBase() {
+  @Autowired
+  lateinit var alertTypeRepository: AlertTypeRepository
+
+  @Autowired
+  lateinit var alertCodeRepository: AlertCodeRepository
+
+  @BeforeEach
+  fun setup() {
+    alertTypeRepository.saveAndFlush(alertTypeVulnerability())
```

Comment:
This alert type is already in the reference data so you could delete this line? I tested it locally to confirm

Reply:
Done.